presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -853,11 +846,9 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                                 source.getProperties());
                     }
                     else {
-                        source = withDerivedProperties(
-                                partitionedExchange(idAllocator.getNextId(), REMOTE, source.getNode(), sourceSymbols, Optional.empty()),
-                                source.getProperties());
+                        source = withDerivedProperties(createPartitionedRemoteExchange(source.getNode(), sourceSymbols, Optional.empty()), source.getProperties());
```

Comment:
I prefer the original format, some line breaks would help read / review. ditto for others :)